### PR TITLE
[Improve] ddl schema column length improvement

### PR DESCRIPTION
**Pull request by wolfboys**
[Improve] ddl schema column length improvement

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @macksonmu @Al-assad PTAL, thx

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql [hunk 1]
```diff
@@ -57,16 +57,16 @@ create table `t_flink_app` (
...
   `app_type` tinyint default null,
   `duration` bigint default null,
   `job_id` varchar(64) collate utf8mb4_general_ci default null,
   `job_manager_url` varchar(255) collate utf8mb4_general_ci default null,
   `version_id` bigint default null,
-  `cluster_id` varchar(255) collate utf8mb4_general_ci default null,
-  `k8s_namespace` varchar(255) collate utf8mb4_general_ci default null,
-  `flink_image` varchar(255) collate utf8mb4_general_ci default null,
-  `state` varchar(50) collate utf8mb4_general_ci default null,
+  `cluster_id` varchar(64) collate utf8mb4_general_ci default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql [hunk 2]
```diff
@@ -57,16 +57,16 @@ create table `t_flink_app` (
...
+  `k8s_namespace` varchar(64) collate utf8mb4_general_ci default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/mysql-schema.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql [hunk 3]
```diff
@@ -205,15 +205,15 @@ create table "public"."t_flink_app" (
...
+  "app_id" varchar(64) collate "pg_catalog"."default",
   "app_type" int2,
   "duration" int8,
   "job_id" varchar(64) collate "pg_catalog"."default",
   "job_manager_url" varchar(255) collate "pg_catalog"."default",
   "version_id" int8,
-  "cluster_id" varchar(255) collate "pg_catalog"."default",
-  "k8s_namespace" varchar(255) collate "pg_catalog"."default",
-  "flink_image" varchar(255) collate "pg_catalog"."default",
+  "cluster_id" varchar(64) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql [hunk 4]
```diff
@@ -205,15 +205,15 @@ create table "public"."t_flink_app" (
...
+  "k8s_namespace" varchar(64) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql [hunk 5]
```diff
@@ -274,17 +271,17 @@ create sequence "public"."streampark_t_flink_cluster_id_seq"
 create table "public"."t_flink_cluster" (
   "id" int8 not null default nextval('streampark_t_flink_cluster_id_seq'::regclass),
   "address" varchar(150) collate "pg_catalog"."default",
-  "cluster_id" varchar(100) collate "pg_catalog"."default",
-  "cluster_name" varchar(100) collate "pg_catalog"."default" not null,
+  "cluster_id" varchar(128) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/schema/pgsql-schema.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 6]
```diff
@@ -36,18 +36,92 @@ create table `t_yarn_queue` (
...
-alter table `t_flink_savepoint` modify column `path` varchar(1024) collate utf8mb4_general_ci default null;
+-- type change
+alter table `t_app_backup` modify column `path` varchar(128) collate utf8mb4_general_ci default null;
+
+alter table `t_flink_app`
+    change column `launch` `release` tinyint default 1,
+    modify column `project_id` bigint default null,
+    modify column `app_id` varchar(64) collate utf8mb4_general_ci default null,
+    modify column `cluster_id` varchar(64) collate utf8mb4_general_ci default null,
+    modify column `k8s_namespace` varchar(64) collate utf8mb4_general_ci default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 7]
```diff
@@ -36,18 +36,92 @@ create table `t_yarn_queue` (
...
-alter table `t_flink_log` add column `option_name` tinyint default null;
-alter table `t_flink_savepoint` modify column `path` varchar(1024) collate utf8mb4_general_ci default null;
+-- type change
+alter table `t_app_backup` modify column `path` varchar(128) collate utf8mb4_general_ci default null;
+
+alter table `t_flink_app`
+    change column `launch` `release` tinyint default 1,
+    modify column `project_id` bigint default null,
+    modify column `app_id` varchar(64) collate utf8mb4_general_ci default null,
+    modify column `cluster_id` varchar(64) collate utf8mb4_general_ci default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 8]
```diff
@@ -36,18 +36,92 @@ create table `t_yarn_queue` (
...
+    modify column `nick_name` varchar(64) collate utf8mb4_general_ci not null comment 'nick name',
+    modify column `salt` varchar(32) collate utf8mb4_general_ci default null comment 'salt',
+    modify column `password` varchar(64) collate utf8mb4_general_ci not null comment 'password',
+    modify column `email` varchar(64) collate utf8mb4_general_ci default null comment 'email',
+    modify column `description` varchar(255) collate utf8mb4_general_ci default null comment 'description',
+    add column `login_type` tinyint default 0 after `user_type`,
+    drop column `avatar`;
+
+alter table `t_flink_cluster`
+    modify column `cluster_id` varchar(128) default null comment 'clusterid of session mode(yarn-session:application-id,k8s-session:cluster-id)',
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 9]
```diff
@@ -36,18 +36,92 @@ create table `t_yarn_queue` (
...
+    modify column `cluster_name` varchar(128) not null comment 'cluster name',
+    modify column `options` text comment 'json form of parameter collection ',
+    modify column `yarn_queue` varchar(128) default null comment 'the yarn queue where the task is located',
+    modify column `k8s_namespace` varchar(64) default 'default' comment 'k8s namespace',
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 10]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
 
+-- type change
+alter table "public"."t_app_backup" modify column "path" varchar(128) collate utf8mb4_general_ci default null;
+
+alter table "public"."t_flink_app"
+    rename column "launch" to "release",
+    alter column "state" TYPE int4 USING state::int4;
+    alter column "app_id" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "cluster_id" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "k8s_namespace" TYPE varchar(64) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 11]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
 drop table if exists "public"."t_flink_tutorial";
 
+-- type change
+alter table "public"."t_app_backup" modify column "path" varchar(128) collate utf8mb4_general_ci default null;
+
+alter table "public"."t_flink_app"
+    rename column "launch" to "release",
+    alter column "state" TYPE int4 USING state::int4;
+    alter column "app_id" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "cluster_id" TYPE varchar(64) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 12]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
+    alter column "nick_name" TYPE varchar(64) collate "pg_catalog"."default" not null,
+    alter column "salt" TYPE varchar(32) collate "pg_catalog"."default"  not null,
+    alter column "password" TYPE varchar(64) collate "pg_catalog"."default" not null,
+    alter column "email" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "description" TYPE varchar(255) collate "pg_catalog"."default",
+    add column "login_type" TYPE int2 default 0;
+    drop column "avatar";
+
+alter table "public"."t_flink_cluster"
+    alter column "cluster_id" TYPE varchar(128) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 13]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
+    alter column "cluster_name" TYPE varchar(128) collate "pg_catalog"."default" not null,
+    alter column "yarn_queue" TYPE varchar(128) collate "pg_catalog"."default",
+    alter column "k8s_namespace" TYPE varchar(64) collate "pg_catalog"."default",
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 14]
```diff
@@ -50,16 +50,16 @@ create table if not exists `t_flink_app` (
...
   `duration` bigint default null,
   `job_id` varchar(64)  default null,
   `job_manager_url` varchar(255)  default null,
   `version_id` bigint default null,
-  `cluster_id` varchar(255)  default null,
-  `k8s_namespace` varchar(255)  default null,
-  `flink_image` varchar(255)  default null,
-  `state` varchar(50)  default null,
+  `cluster_id` varchar(64)  default null,
+  `k8s_namespace` varchar(64)  default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 15]
```diff
@@ -50,16 +50,16 @@ create table if not exists `t_flink_app` (
...
   `app_type` tinyint default null,
   `duration` bigint default null,
   `job_id` varchar(64)  default null,
   `job_manager_url` varchar(255)  default null,
   `version_id` bigint default null,
-  `cluster_id` varchar(255)  default null,
-  `k8s_namespace` varchar(255)  default null,
-  `flink_image` varchar(255)  default null,
-  `state` varchar(50)  default null,
+  `cluster_id` varchar(64)  default null,
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Change to varchar(45)

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 16]
```diff
@@ -373,17 +372,17 @@ create table if not exists `t_app_build_pipe` (
...
+  `cluster_id` varchar(128) default null comment 'clusterId of session mode(yarn-session:application-id,k8s-session:cluster-id)',
+  `cluster_name` varchar(128) not null comment 'cluster name',
   `options` text comment 'json form of parameter collection ',
-  `yarn_queue` varchar(100) default null comment 'the yarn queue where the task is located',
+  `yarn_queue` varchar(128) default null comment 'the yarn queue where the task is located',
   `execution_mode` tinyint not null default 1 comment 'k8s execution session mode(1:remote,3:yarn-session,5:kubernetes-session)',
   `version_id` bigint not null comment 'flink version id',
-  `k8s_namespace` varchar(255) default 'default' comment 'k8s namespace',
-  `service_account` varchar(50) default null comment 'k8s service account',
+  `k8s_namespace` varchar(64) default 'default' comment 'k8s namespace',
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Change to varchar(63)

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 17]
```diff
@@ -373,17 +372,17 @@ create table if not exists `t_app_build_pipe` (
 create table if not exists `t_flink_cluster` (
   `id` bigint  generated by default as identity not null,
   `address` varchar(150) default null comment 'url address of jobmanager',
-  `cluster_id` varchar(100) default null comment 'clusterId of session mode(yarn-session:application-id,k8s-session:cluster-id)',
-  `cluster_name` varchar(100) not null comment 'cluster name',
+  `cluster_id` varchar(128) default null comment 'clusterId of session mode(yarn-session:application-id,k8s-session:cluster-id)',
```
Comment (Al-assad) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Change to varchar(45)

**Review by Al-assad — COMMENTED** [review]
Hi @wolfboys , I think the length of some of the fields related to k8s resources should be fixed.

The k8s namespace resource requires compliance with the RFC-1123 label names standard (contains at most 63 characters).
ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#namespaces-and-dns

Flink kubernetes.cluster-id configuration item is limited to a maximum length of 45 characters.
ref: https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#kubernetes-cluster-id

File: streampark-console/streampark-console-service/src/main/assembly/script/data/mysql-data.sql [hunk 18]
```diff
@@ -135,8 +135,8 @@ insert into `t_menu` values (130603, 130600, 'delete yarn queue', null, null, 'y
 -- ----------------------------
 -- Records of t_role
 -- ----------------------------
-insert into `t_role` values (100001, 'developer', 'developer', now(), now(), null);
-insert into `t_role` values (100002, 'team admin', 'Team Admin has all permissions inside the team.', now(), now(), null);
+insert into `t_role` values (100001, 'developer', now(), now(), 'developer');
+insert into `t_role` values (100002, 'team admin', now(), now(), 'Team Admin has all permissions inside the team.', );
```
Comment (muchunjin) [inline streampark-console/streampark-console-service/src/main/assembly/script/data/mysql-data.sql]: There is an extra comma after the last column value

File: streampark-console/streampark-console-service/src/main/assembly/script/data/pgsql-data.sql [hunk 19]
```diff
@@ -131,8 +131,8 @@ insert into "public"."t_menu" values (130603, 130600, 'delete yarn queue', null,
 -- ----------------------------
 -- Records of t_role
 -- ----------------------------
-insert into "public"."t_role" values (100001, 'developer', 'developer', now(), now(), null);
-insert into "public"."t_role" values (100002, 'team admin', 'Team Admin has all permissions inside the team.', now(), now(), null);
+insert into "public"."t_role" values (100001, 'developer', now(), now(), 'developer');
+insert into "public"."t_role" values (100002, 'team admin', now(), now(), 'Team Admin has all permissions inside the team.', );
```
Comment (muchunjin) [inline streampark-console/streampark-console-service/src/main/assembly/script/data/pgsql-data.sql]: There is an extra comma after the last column value

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 20]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
 alter table "public"."t_yarn_queue" add constraint "unique_team_id_queue_label" unique("team_id", "queue_label");
-alter table "public"."t_flink_log" add column "option_name" int2;
-alter table "public"."t_user" add column "login_type" int2 default 0;
-ALTER TABLE public.t_flink_app ALTER COLUMN state TYPE int4 USING state::int4;
+
 
 drop table if exists "public"."t_flink_tutorial";
 
+-- type change
+alter table "public"."t_app_backup" modify column "path" varchar(128) collate utf8mb4_general_ci default null;
```
Comment (muchunjin) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: Modify to `alter table "public"."t_app_backup" alter column "path" TYPE varchar(128) collate "pg_catalog"."default";`

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 21]
```diff
@@ -54,12 +52,93 @@ comment on column "public"."t_yarn_queue"."modify_time" is 'modify time';
...
+
+alter table "public"."t_flink_app"
+    rename column "launch" to "release",
+    alter column "state" TYPE int4 USING state::int4;
+    alter column "app_id" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "cluster_id" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "k8s_namespace" TYPE varchar(64) collate "pg_catalog"."default",
+    alter column "flink_image" TYPE varchar(128) collate "pg_catalog"."default";
```
Comment (muchunjin) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql:69]: alter table "public"."t_flink_app"
    rename column "launch" to "release",
    alter column "state" TYPE int4 USING state::int4;
    alter column "app_id" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "cluster_id" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "k8s_namespace" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "flink_image" TYPE varchar(128) collate "pg_catalog"."default";

`modify to`

alter table "public"."t_flink_app" rename column "launch" to "release";

alter table "public"."t_flink_app"
    alter column "state" TYPE int4 USING state::int4,
    alter column "app_id" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "cluster_id" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "k8s_namespace" TYPE varchar(64) collate "pg_catalog"."default",
    alter column "flink_image" TYPE varchar(128) collate "pg_catalog"."default";

**Comment by wolfboys** [comment]
@Al-assad @macksonmu  Thanks for your review, updated, please review again.

**Comment by wolfboys** [comment]
> Hi @wolfboys , I think the length of some of the fields related to k8s resources should be fixed.
> 
> The k8s namespace resource requires compliance with the RFC-1123 label names standard (contains at most 63 characters). ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#namespaces-and-dns
> 
> Flink kubernetes.cluster-id configuration item is limited to a maximum length of 45 characters. ref: https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#kubernetes-cluster-id

Thanks for your review, updated.

**Review by Al-assad — DISMISSED** [review]
It looks good to me now. 😃

**Review by muchunjin — DISMISSED** [review]
LGTM

**Review by muchunjin — APPROVED** [review]
LGTM

**Pull request merged**